src/SW_Main_lib.c

```diff
@@ -191,19 +191,98 @@ void sw_init_args(int argc, char **argv, Bool *QuietMode,
 
 }
 
+/**
+ * @brief In SOILWAT2 and rSOILWAT2, if an error occurs, this function
+ *  notifies the user and crashes
+ *
+ * @param[in] QuietMode Flag to control if the program version is displayed
+ * @param[in] LogInfo Holds information dealing with logfile output
+*/
+void sw_check_exit(Bool QuietMode, LOG_INFO* LogInfo) {
+    #ifdef RSOILWAT
+    if(LogInfo->stopRun) {
+        error(LogInfo->errorMsg);
+    }
+    #else
+    if(LogInfo->stopRun && !QuietMode) {
+        fprintf(stderr, "%s", LogInfo->errorMsg);
+        exit(-1);
+    }
+    #endif
+}
 
-void sw_check_log(Bool QuietMode, LOG_INFO* LogInfo) {
-	/* =================================================== */
-	/* function to be called by atexit() so it's the last
-	 * to execute before termination.  This is the place to
-	 * do any cleanup or progress reporting.
-	 */
-	if (LogInfo->logfp != stdout && LogInfo->logfp != stderr) {
-		CloseFile(&LogInfo->logfp, LogInfo);
-		if (LogInfo->logged && !QuietMode) {
-			fprintf(stderr, "\nCheck logfile for error or status messages.\n");
-			fflush(stderr);
-		}
+/**
+ * @brief Initialize values within LOG_INFO
+ *
+ * @param[in] logInitPtr Log file pointer that LOG_INFO will be initialized to
+ * @param[in,out] LogInfo Holds information dealing with log file output
+*/
+void sw_init_logs(FILE* logInitPtr, LOG_INFO* LogInfo) {
+
+	LogInfo->logged = swFALSE;
+	LogInfo->logfp = logInitPtr;
+
+	LogInfo->errorMsg[0] = '\0';
+
+	LogInfo->stopRun = swFALSE;
+	LogInfo->numWarnings = 0;
+}
+
+/**
+ * @brief Write logs that have been accumulated throughout the program/
+ * 	simulation run
+ *
+ * @param[in] QuietMode Flag to control if the program version is displayed
+ * @param[in] LogInfo Holds information dealing with logfile output
+*/
+void sw_write_logs(Bool QuietMode, LOG_INFO* LogInfo) {

```

Comment:
In `sw_write_logs()` if `RSOILWAT`, we currently call `warning()` if we got a warning message - previously, we had functionality (in `LogError()`) that didn't create warning messages if the user requested silence (see old code comment "// send warning message only if not silenced (fp is not NULL)". Please add corresponding functionality back in.